internal/exec/exec.go

```diff
@@ -250,16 +251,19 @@ func execFieldSelection(ctx context.Context, r *Request, s *resolvable.Schema, f
 				res = res.Elem()
 			}
 
-			res, resolverErr := r.executeWithDirectives(ctx, f, nil, func(ctx context.Context, args interface{}) (output interface{}, err error) {
+			resolver := func(ctx context.Context, _ interface{}) (output interface{}, err error) {
 				return res.FieldByIndex(f.field.FieldIndex).Interface(), nil
-			})
+			}
+
+			res, resolverErr := r.executeWithDirectives(ctx, f, nil, resolver)
 			if resolverErr != nil {
 				err := errors.Errorf("%s", resolverErr)
 				err.Path = path.toSlice()
 				err.ResolverError = resolverErr
 				if ex, ok := resolverErr.(extensionser); ok {
 					err.Extensions = ex.Extensions()
 				}
+				return err
 			}
```

Comment:
Now this entire block can be taken out of the `if`/`else` statement because it is pretty much the same.

Reply:
Moved to a receiver function on `fieldToExec`, and which removes the duplication and IMO improves the readability/scope